projects/kit/src/lib/utils/date/to-date-string.ts

```diff
@@ -11,11 +11,18 @@ export function toDateString(
         seconds,
         milliseconds,
     }: Partial<MaskitoDateSegments & Partial<MaskitoTimeSegments>>,
-    dateMode: string,
-    timeMode?: string,
+    {
+        dateMode,
+        dateTimeSeparator = DATE_TIME_SEPARATOR,
+        timeMode,
+    }: {
+        dateMode: string;
+        dateTimeSeparator?: string;
+        timeMode?: string;
```

Comment:
`dateTimeSeparator` cannot be used without `timeMode`.
And `timeMode` cannot be used without `dateTimeSeparator`.

I think it may be useful to add these overloads

```ts
export function toDateString(
    segments: Partial<MaskitoDateSegments>,
    options: {
        dateMode: string;
    },
): string;

export function toDateString(
    segments: Partial<MaskitoDateSegments> & Partial<MaskitoTimeSegments>,
    options: {
        dateMode: string;
        timeMode: string;
        dateTimeSeparator: string;
    },
): string;
```